students/brandon_nguyen/session08/Circle.py

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env python3
+# Week8 Excercise circle.py
+# Student: Brandon Nguyen - Au2018
+
+from math import pi
+
+
+class Circle(object):
+
+    def __init__(self, radius):
+        # check radius error r !<= 0
+        if radius <= 0:
+            raise ValueError("Radius is defined to be greater than 0!")
+        # self.radius = radius  # let use @property instead
+        self.__radius = radius
+        self.__diameter = radius * 2
+
+    @property
+    def radius(self):
+        return self.__radius
+
+    @radius.setter
+    # now we can reset the radius & diameter
+    def radius(self, r):
+        self.__radius = r
+        self.__diameter = r*2
+
+    @property
+    def diameter(self):
+        return self.__diameter
+    
+    @diameter.setter
+    def diameter(self, d):
+        self.__radius = d/2
```

Comment:
You should also update the `self.__diameter` as well. Otherwise, your `self.__radius` and `self.__diameter` might be inconsistent after `c = Circle(5)` and `c.diameter = 4` is executed (for example).

In fact, a better approach might be just to remember radius only, and derive diameter in the property getter, not remembering diameter separately.

Reply:
thank you! will do.